signer/tuf_on_ci_sign/_signer_repository.py

```diff
@@ -97,6 +99,15 @@ def _find_changed_roles(known_good_dir: str, signing_event_dir: str) -> list[str
     return changed_roles
 
 
+def set_key_field(key: Key, name: str, value: str) -> None:
+    key.unrecognized_fields[f"x-tuf-on-ci-{name}"] = value
+    # This is dumb but TUF spec requires the keyid to change at this point
```

Comment:
This is dumb indeed and ambiguous too. As one could argue that unrecognised fields are not part of the "canonical form" of the key.